services/cd-service/pkg/repository/transformer.go

```diff
@@ -1649,7 +1750,20 @@ func (c *DeployApplicationVersion) Transform(
 		if err != nil {
 			return "", err
 		}
-		if len(envLocks) > 0 || len(appLocks) > 0 {
+
+		appDir := applicationDirectory(fs, c.Application)
+
+		team, err := util.ReadFile(fs, fs.Join(appDir, "team"))
+
+		if errors.Is(err, os.ErrNotExist) {
+			teamLocks = map[string]Lock{} //This is a workaround. This transformer now needs a team file to be created on the application folder in order for the team locks to be loaded correctly. It turns out almost no other test that uses the CreateApplicationVersion transformer (that uses this one) sets a value for the team parameter, so the file wont exist and the tests will fail. IF we can't find the file, simply create an empty map and move on.
```

Comment:
this is not just a test issue, but will always happen when we create the first lock for a given team.
So I don't think this is a "workaround". If there's no directory, then there is no lock, that's working as expected.